go.mod

```
@@ -7,14 +7,14 @@ require (
 	github.com/gobuffalo/flect v0.2.3
 	github.com/joelanford/go-apidiff v0.1.0
 	github.com/onsi/ginkgo v1.16.5
-	github.com/onsi/gomega v1.17.0
+	github.com/onsi/gomega v1.18.1
 	github.com/sirupsen/logrus v1.8.1
 	github.com/spf13/afero v1.6.0
-	github.com/spf13/cobra v1.2.1
+	github.com/spf13/cobra v1.4.0
 	github.com/spf13/pflag v1.0.5
-	golang.org/x/tools v0.1.6-0.20210820212750-d4cc65f0b2ff
-	k8s.io/apimachinery v0.23.5 // for `kubebuilder alpha config-gen`
-	sigs.k8s.io/controller-runtime v0.11.2
+	golang.org/x/tools v0.1.10-0.20220218145154-897bd77cd717
+	k8s.io/apimachinery v0.24.0 // for `kubebuilder alpha config-gen`
+	sigs.k8s.io/controller-runtime v0.12.1
 	sigs.k8s.io/controller-tools v0.8.0 // for `kubebuilder alpha config-gen`
```

controller-tools should be upgraded to `v0.9.0` since it doesn't support k8s 1.24 until that release. More info on the v0.9.0 release can be found here: https://github.com/kubernetes-sigs/controller-tools/releases/tag/v0.9.0

```suggestion
	sigs.k8s.io/controller-tools v0.9.0 // for `kubebuilder alpha config-gen`
```

---

@NikhilSharmaWe, 
We need also add this info in the title and description
I am doing tha